tests/--inadvisable-executable-name--

```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+echo hello
```

Comment:
How about generating this dynamically rather than shipping it in the source to be nice to other tools that might process source code and not be prepared to handle such a filename?  (Some license checker, sloccount, etc.)